### PR TITLE
New view options

### PR DESCRIPTION
**Pull request by brandelune**
This PR keeps the view options related commits from https://github.com/omegat-org/omegat/pull/386

### CODE REVIEW

**Review by miurahr — CHANGES_REQUESTED** [review]
It should place default constants for editor in `EditorSettings` class as `private static final boolean`, not in `Preferences` class.
It is also a good habit to put implicit defaults for other settings.

**Comment by brandelune** [comment]
I don't understand what you are asking me to change.

File: src/org/omegat/gui/editor/EditorSettings.java [hunk 1]
```diff
@@ -83,24 +83,28 @@ protected EditorSettings(final EditorController parent) {
         markTranslated = Preferences.isPreference(Preferences.MARK_TRANSLATED_SEGMENTS);
         markUntranslated = Preferences.isPreference(Preferences.MARK_UNTRANSLATED_SEGMENTS);
         displaySegmentSources = Preferences.isPreference(Preferences.DISPLAY_SEGMENT_SOURCES);
-        markNonUniqueSegments = Preferences.isPreference(Preferences.MARK_NON_UNIQUE_SEGMENTS);
+        markNonUniqueSegments = Preferences.isPreferenceDefault(Preferences.MARK_NON_UNIQUE_SEGMENTS,
+                Preferences.MARK_NON_UNIQUE_SEGMENTS_DEFAULT);
```
Comment (miurahr) [inline src/org/omegat/gui/editor/EditorSettings.java:87]:
```suggestion
               MARK_NON_UNIQUE_SEGMENTS_DEFAULT);
```

File: src/org/omegat/gui/editor/EditorSettings.java [hunk 2]
```diff
@@ -83,24 +83,28 @@ protected EditorSettings(final EditorController parent) {
...
 
         // options from preferences 'view' pane
         viewSourceBold = Preferences.isPreferenceDefault(Preferences.VIEW_OPTION_SOURCE_ALL_BOLD,
                 Preferences.VIEW_OPTION_SOURCE_ALL_BOLD_DEFAULT);
         viewActiveSourceBold = Preferences.isPreferenceDefault(Preferences.VIEW_OPTION_SOURCE_ACTIVE_BOLD,
                 Preferences.VIEW_OPTION_SOURCE_ACTIVE_BOLD_DEFAULT);
         markFirstNonUnique = Preferences.isPreference(Preferences.VIEW_OPTION_UNIQUE_FIRST);
-        markGlossaryMatches = Preferences.isPreference(Preferences.MARK_GLOSSARY_MATCHES);
+        markGlossaryMatches = Preferences.isPreferenceDefault(Preferences.MARK_GLOSSARY_MATCHES,
+                Preferences.MARK_GLOSSARY_MATCHES_DEFAULT);
```
Comment (miurahr) [inline src/org/omegat/gui/editor/EditorSettings.java:107]:
```suggestion
                MARK_GLOSSARY_MATCHES_DEFAULT);
```

File: src/org/omegat/util/Preferences.java [hunk 3]
```diff
@@ -192,6 +192,7 @@ public final class Preferences {
      * backwards-compatibility.
      */
     public static final String MARK_GLOSSARY_MATCHES = "transtips";
+    public static final boolean MARK_GLOSSARY_MATCHES_DEFAULT = true;
```
Comment (miurahr) [inline src/org/omegat/util/Preferences.java:195]:
```suggestion

```

File: src/org/omegat/util/Preferences.java [hunk 4]
```diff
@@ -215,7 +217,8 @@ public final class Preferences {
     public static final String MARK_TRANSLATED_SEGMENTS = "mark_translated_segments";
 
     public static final String MARK_AUTOPOPULATED = "mark_autopopulated";
-
+    public static final boolean MARK_AUTOPOPULATED_DEFAULT = true;
```
Comment (miurahr) [inline src/org/omegat/util/Preferences.java:220]:
```suggestion

```

File: src/org/omegat/util/Preferences.java [hunk 5]
```diff
@@ -342,7 +345,8 @@ public final class Preferences {
      * mark unique segments
      */
     public static final String MARK_NON_UNIQUE_SEGMENTS = "mark_non_unique_segments";
-
+    public static final boolean MARK_NON_UNIQUE_SEGMENTS_DEFAULT = true;
```
Comment (miurahr) [inline src/org/omegat/util/Preferences.java:348]:
```suggestion

```

**Review by miurahr — CHANGES_REQUESTED** [review]

**Comment by brandelune** [comment]
I still don't understand, but I did it.

**Comment by miurahr** [comment]
Reviewer cannot add a new code. You should add a constant variable definition in the class.

**Comment by brandelune** [comment]
> It should place default constants for editor in `EditorSettings` class as `private static final boolean`, not in `Preferences` class.
> It is also a good habit to put implicit defaults for other settings.

If we split preference settings in all the related files, don't you think it makes it more difficult to find them ?

**Comment by miurahr** [comment]
> > It should place default constants for editor in `EditorSettings` class as `private static final boolean`, not in `Preferences` class.
> > It is also a good habit to put implicit defaults for other settings.
> 
> If we split preference settings in all the related files, don't you think it makes it more difficult to find them ?

I don't understand what you want to do.

There is a golden rule for object-oriented programming.
All the related properties, methods are going to be inside the class.

I think as follows based on the OOP rule.
Keys of preferences are properties (constant) of Preferences class. 
The default values (constant ) of  Editor behavior  are properties of  Editor GUI parts, so it should  belong to `EditorSettings` class.

I think that finding variable is a task of IDE, cross reference tools, or other computer-assisted programming environment, especially in large project in Java ecosystem.

**Comment by brandelune** [comment]
> I don't understand what you want to do.
> 
> There is a golden rule for object-oriented programming. All the related properties, methods are going to be inside the class.

I'm fine with golden rules, but I'm just changing 5 default preferences in the _current_ version of the code.

> I think as follows based on the OOP rule. Keys of preferences are properties (constant) of Preferences class. The default values (constant ) of Editor behavior are properties of Editor GUI parts, so it should belong to `EditorSettings` class.

If we want to reorganize the whole preference code to be closer to the traditional OOP rules, we can do that in 6.1.

**Comment by miurahr** [comment]
> > I don't understand what you want to do.
> > There is a golden rule for object-oriented programming. All the related properties, methods are going to be inside the class.
> 
> I'm fine with golden rules, but I'm just changing 5 default preferences in the _current_ version of the code.
> 
> > I think as follows based on the OOP rule. Keys of preferences are properties (constant) of Preferences class. The default values (constant ) of Editor behavior are properties of Editor GUI parts, so it should belong to `EditorSettings` class.
> 
> If we want to reorganize the whole preference code to be closer to the traditional OOP rules, we can do that in 6.1.

I just comment about your 5 default preferences. I don't understand why you tell whole preference code.

**Comment by brandelune** [comment]
Sorry, maybe there is a confusion.

Is there any problem with the current proposals ?

**Comment by miurahr** [comment]
There is no big problem in current code; except for failure of build.


I want to understand a reason why default value should be in Preferences class.

In many cases,  OmegaT code use direct value for defaults. Some put its default as  class constant,  and very rare who put it in Preferences class.

It may be a good change to improve Preferences system when we find something not optimal in the system.

There can be an idea to put defaults values in external file such as `PreferenesDefaults.properties` and developer and manual editor can share its values. This may help document writer to understand  behaviors.
Is it your concerned?

**Comment by brandelune** [comment]
> There is no big problem in current code; except for failure of build.

When I merge master on this branch it builds fine.
 
> I want to understand a reason why default value should be in Preferences class.

I see.
 
> In many cases, OmegaT code use direct value for defaults. Some put its default as class constant, and very rare who put it in Preferences class.

I see.
 
> It may be a good change to improve Preferences system when we find something not optimal in the system.

Indeed.

> There can be an idea to put defaults values in external file such as `PreferenesDefaults.properties` and developer and manual editor can share its values. This may help document writer to understand behaviors. Is it your concerned?

Yes.

But what matters more than may concern is that people who are involved in development or who want to be involved in development find that modifying preferences is easy.

So, if there is a better (more systematic) way to implement preferences than what exists now we should consider it.

**Review by miurahr — APPROVED** [review]

**Comment by miurahr** [comment]
I've squashed and merged.

**Pull request closed without merging**